packages/fast-colors/src/colorlib/colorConverters.ts

```diff
@@ -0,0 +1,385 @@
+        }
+    }
+
+    const x: number = xyzToLABHelper(xyz.x / ColorXYZ.whitePoint.x);
+    const y: number = xyzToLABHelper(xyz.y / ColorXYZ.whitePoint.y);
+    const z: number = xyzToLABHelper(xyz.z / ColorXYZ.whitePoint.z);
+
+    const l: number = 116 * y - 16;
+    const a: number = 500 * (x - y);
+    const b: number = 200 * (y - z);
+
+    return new ColorLAB(l, a, b);
+}
+
+// The alpha channel of the input is ignored
+export function rgbToXYZ(rgb: ColorRGBA64): ColorXYZ {
+    function rgbToXYZHelper(i: number): number {
+        if (i <= 0.04045) {
+            return i / 12.92;
+        } else {
+            return Math.pow((i + 0.055) / 1.055, 2.4);
+        }
+    }
+
+    const r: number = rgbToXYZHelper(rgb.r);
+    const g: number = rgbToXYZHelper(rgb.g);
+    const b: number = rgbToXYZHelper(rgb.b);
+
+    const x: number = r * 0.4124564 + g * 0.3575761 + b * 0.1804375;
+    const y: number = r * 0.2126729 + g * 0.7151522 + b * 0.072175;
+    const z: number = r * 0.0193339 + g * 0.119192 + b * 0.9503041;
+
+    return new ColorXYZ(x, y, z);
+}
+
+// Note that the xyz color space is signifigantly larger than sRGB. As such, this can return colors rgb values greater than 1 or less than 0
+export function xyzToRGB(xyz: ColorXYZ, alpha: number = 1): ColorRGBA64 {
+    function xyzToRGBHelper(i: number): number {
+        if (i <= 0.0031308) {
+            return i * 12.92;
+        } else {
+            return 1.055 * Math.pow(i, 1 / 2.4) - 0.055;
+        }
+    }
+
+    const r: number = xyzToRGBHelper(
+        xyz.x * 3.2404542 - xyz.y * 1.5371385 - xyz.z * 0.4985314
+    );
+    const g: number = xyzToRGBHelper(
+        xyz.x * -0.969266 + xyz.y * 1.8760108 + xyz.z * 0.041556
+    );
+    const b: number = xyzToRGBHelper(
+        xyz.x * 0.0556434 - xyz.y * 0.2040259 + xyz.z * 1.0572252
+    );
+
+    return new ColorRGBA64(r, g, b, alpha);
+}
+
+// The alpha channel of the input is ignored
+export function rgbToLAB(rgb: ColorRGBA64): ColorLAB {
+    return xyzToLAB(rgbToXYZ(rgb));
+}
+
+// Note that the xyz color space (which the conversion from LAB uses) is signifigantly larger than sRGB. As such, this can return colors rgb values greater than 1 or less than 0
+export function labToRGB(lab: ColorLAB, alpha: number = 1): ColorRGBA64 {
+    return xyzToRGB(labToXYZ(lab), alpha);
+}
+
+// The alpha channel of the input is ignored
+export function rgbToLCH(rgb: ColorRGBA64): ColorLCH {
+    return labToLCH(rgbToLAB(rgb));
+}
+
+export function lchToRGB(lch: ColorLCH, alpha: number = 1): ColorRGBA64 {
+    return labToRGB(lchToLAB(lch), alpha);
+}
+
+export function temperatureToRGB(tempKelvin: number, alpha: number = 1): ColorRGBA64 {
+    // The constants I could find assumed a decimal range of [0,255] for each channel. Just going to put a /255.0 at the end
+    let r: number = 0;
+    let g: number = 0;
+    let b: number = 0;
+
+    if (tempKelvin <= 1000) {
+        tempKelvin = 1000;

```

Comment:
We've been avoiding the reassigning values from arguments passed, it almost feels like a side effect from running this function that tempKelvin is normalized to 1000 or 40000. Whats the opinion on this and if we go forward I suggest we document what this will do to the passed argument as it's not what I would have expected.

Reply:
@nicholasrice @eigenvector32 The other comments I've made are just nits, but this one I would like to address.